src/test/java/com/smartbear/readyapi/client/DataGenDataSourceTest.java

```diff
@@ -0,0 +1,44 @@
+package com.smartbear.readyapi.client;
+
+import com.smartbear.readyapi.client.model.BooleanDataGenerator;
+import com.smartbear.readyapi.client.model.DataGenDataSource;
+import com.smartbear.readyapi.client.model.DataSourceTestStep;
+import com.smartbear.readyapi.client.model.TestStep;
+import com.smartbear.readyapi.client.teststeps.TestStepTypes;
+import com.smartbear.readyapi.client.teststeps.datasource.datagen.BooleanDataGeneratorBuilder;
+import org.junit.Test;
+
+import static com.smartbear.readyapi.client.TestRecipeBuilder.newTestRecipe;
+import static com.smartbear.readyapi.client.teststeps.TestSteps.dataGenDataSource;
+import static com.smartbear.readyapi.client.teststeps.datasource.datagen.DataGenerators.booleanDataGenerator;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class DataGenDataSourceTest {
+
+    @Test
+    public void buildsRecipeWithDataSourceTestStepWithBooleanDataGenDataSource() throws Exception {
+        TestRecipe recipe = newTestRecipe()
+                .addStep(dataGenDataSource()
+                        .setNumberOfRows(34)
+                        .addDataGenerator(
+                                booleanDataGenerator("property1")
```

Comment:
I like 'booleanProperty("property1").duplicatedBy( 1 ).withYesNoFormat() ', Duplicated by is optional, default is 1 - so I think it's better if we can keep the api also that way so that user can skip it if he wants to use default value.

Should I change to this or do you think we should go for the other suggestions?